src/test/run-make/linker-output-non-utf8/Makefile

```diff
@@ -2,15 +2,12 @@
 
 # Make sure we don't ICE if the linker prints a non-UTF-8 error message.
 
-ifdef IS_WINDOWS
-# ignore windows
+# Ignore Windows and Apple
 
-# This does not work in its current form on windows, possibly due to
-# gcc bugs or something about valid Windows paths.  See issue #29151
-# for more information.
-all:
-
-else
+# This does not work in its current form on Windows or Apple APFS due
+# to their filesystems requiring paths to be valid Unicode.
```

Comment:
Are you sure Windows require paths to be valid Unicode? If not, please retain the original comment about Windows, and open a new sentence about the valid Unicode requirement with APFS.

Reply:
Yes, it needs to parse the path as Unicode to perform normalisation and case folding on NTFS and ReFS. [MSDN](https://msdn.microsoft.com/en-us/library/windows/desktop/dd317748(v=vs.85).aspx) covers this briefly

Reply:
This is not true - when MSDN refers to "unicode" it means un-validated UCS-2 - case mapping ignores invalid sequences. @retep998 probably knows more details.

Reply:
Windows does not perform unicode normalization on paths. Case folding is also done via a highly simplistic `u16` to `u16` mapping which will handle lone surrogates just fine. Windows uses WTF-16 for paths, which is a superset of UTF-16.

Reply:
Thanks for the correction. I'll switch the comment back.